### PR TITLE
Deprecated ContainerAwareEventDispatcher

### LINKED ISSUE #7340: ContainerAwareEventDispatcher has been deprecated
Use `EventDispatcher` with closure-proxy injection instead.

See https://github.com/symfony/symfony/pull/20937

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #7340.

### CODE REVIEW

**Comment by xabbuh** [comment]
Instead of documenting the deprecation I would update the example to show code that will be working with Symfony 4.0 too.

**Comment by javiereguiluz** [comment]
@xabbuh you mean the example in `components/event_dispatcher.rst`?

**Comment by xabbuh** [comment]
Yes, maybe move it to its own article and then also add a link in the existing chapter to it.

**Comment by javiereguiluz** [comment]
I wonder ... why not remove that example entirely?  It doesn't explain nothing special ... just how to define services with tags, right?

**Comment by xabbuh** [comment]
Fine for me too. I am just afraid that users with older Symfony versions might be confused (sure, in an ideal world they wouldn't read the 3.2 docs, but you know what's going on).

**Comment by javiereguiluz** [comment]
I've checked other articles and I have a question.

In `service_container/tags.rst` we say that if you add this code, the Twig extension will be automatically registered:

```php
$definition = new Definition(AppExtension::class);
$definition->setPublic(false);
$definition->addTag('twig.extension');
$container->setDefinition('app.twig_extension', $definition);
```

But that's not true, right? You need to add some special compiler pass to manage the `twig.extension` tag.

That's why the example in `event_dispatcher.rst` mentions the `RegisterListenersPass()`.

---

So I propose the following:

* Remove the `event_dispatcher.rst` example entirely
* Improve `service_container/tags.rst` to better explain what you must do to enable the service tags

**Comment by xabbuh** [comment]
Well, `service_container/tags.rst` addresses users of the full-stack framework so we can safely assume that the compiler pass is registered.

File: components/event_dispatcher.rst [hunk 1]
```diff
@@ -187,33 +187,28 @@ event. In many cases, a special event subclass is passed with extra
...
-    :class:`Symfony\\Component\\EventDispatcher\\ContainerAwareEventDispatcher`
-    and the
-    :doc:`DependencyInjection component </components/dependency_injection>`,
-    you can use the
-    :class:`Symfony\\Component\\EventDispatcher\\DependencyInjection\\RegisterListenersPass`
-    to tag services as event listeners::
+    Registering service definitions and tagging them with the ``kernel.event_listener``
+    and ``kernel.event_subscriber`` tags is not enough to enable the event listeners
+    and event subscribers. You must also register the ``RegisterListenersPass()``
+    compiler pass in the container builder::
```
Comment (xabbuh) [inline components/event_dispatcher.rst]: I would remove the "compiler pass" part here. This reads a bit weird.

File: components/event_dispatcher.rst [hunk 2]
```diff
@@ -189,10 +189,10 @@ determine which instance is passed.
...
 .. sidebar:: Registering Event Listeners and Subscribers in the Service Container
 
-    Registering service definitions and tagging them with the ``kernel.event_listener``
-    and ``kernel.event_subscriber`` tags is not enough to enable the event listeners
-    and event subscribers. You must also register the ``RegisterListenersPass()``
-    compiler pass in the container builder::
+    Registering service definitions and tagging them with the
+    ``kernel.event_listener`` and ``kernel.event_subscriber`` tags is not enough
+    to enable the event listeners and event subscribers. You must also register
+    in the container builder a compiler pass called ``RegisterListenersPass()``::
```
Comment (xabbuh) [inline components/event_dispatcher.rst]: You must also register a compiler pass call `RegisterListenersPass` in the container builder

**Comment by xabbuh** [comment]
👍 

Status: Reviewed

**Comment by xabbuh** [comment]
Thank you @javiereguiluz.

**Pull request closed without merging**